cli/cluster.go

```diff
@@ -337,55 +338,12 @@ func runClusterBackup(args *docopt.Args) error {
 
 	fmt.Fprintln(os.Stderr, "Creating cluster backup...")
 
-	tw := NewTarWriter("flynn-backup-"+time.Now().UTC().Format("2006-01-02_150405"), dest)
-	defer tw.Close()
-
-	// get app and release details for key apps
-	data := make(map[string]*ct.ExpandedFormation, 4)
-	for _, name := range []string{"postgres", "discoverd", "flannel", "controller"} {
-		app, err := client.GetApp(name)
-		if err != nil {
-			return fmt.Errorf("error getting %s app details: %s", name, err)
-		}
-		release, err := client.GetAppRelease(app.ID)
-		if err != nil {
-			return fmt.Errorf("error getting %s app release: %s", name, err)
-		}
-		formation, err := client.GetFormation(app.ID, release.ID)
-		if err != nil {
-			return fmt.Errorf("error getting %s app formation: %s", name, err)
-		}
-		artifact, err := client.GetArtifact(release.ArtifactID)
-		if err != nil {
-			return fmt.Errorf("error getting %s app artifact: %s", name, err)
-		}
-		data[name] = &ct.ExpandedFormation{
-			App:       app,
-			Release:   release,
-			Artifact:  artifact,
-			Processes: formation.Processes,
-		}
+	if bar != nil {
+		dest = io.MultiWriter(dest, bar)
 	}
-	if err := tw.WriteJSON("flynn.json", data); err != nil {
+	if err := clusterbackup.RunBackup(client, dest); err != nil {
```

Comment:
I think the package should be `backup` and the function `Run`, so this becomes `backup.Run`, WDYT?